test/geocoders/geonames.py

```diff
@@ -55,20 +55,62 @@ def test_query_urlencoding(self):
 
     def test_reverse(self):
         self.reverse_run(
-            {"query": "40.75376406311989, -73.98489005863667", "exactly_one": True},
-            {"latitude": 40.75376406311989, "longitude": -73.98489005863667},
+            {
+                "query": "40.75376406311989, -73.98489005863667",
+                "exactly_one": True,
+            },
+            {
+                "latitude": 40.75376406311989,
+                "longitude": -73.98489005863667,
+                "address": "Times Square, NY, US",
+            },
         )
 
-        self.assertRaises(
-            ValueError,
-            self._make_request,
-            self.geocoder.reverse,
-            **{"query": "40.75376406311989, -73.98489005863667",
-               "exactly_one": True,
-               "find_nearby_type": "findNearbyJSON",
-               "lang": 'en'}
+        self.reverse_run(
+            {
+                "query": "40.75376406311989, -73.98489005863667",
+                "exactly_one": True,
+                "find_nearby_type": 'findNearbyJSON',
+            },
+            {
+                "latitude": 40.75376406311989,
+                "longitude": -73.98489005863667,
+                "address": "Bryant Park Studios, NY, US",
+            },
         )
 
+        self.reverse_run(
+            {
+                "query": "40.75376406311989, -73.98489005863667",
+                "exactly_one": True,
+                "find_nearby_type": 'findNearbyJSON',
+                "feature_code": "ADM1"
+            },
+            {
+                "latitude": 40.16706,
+                "longitude": -74.49987,
+            },
+        )
+
+        loc = self._make_request(
+                self.geocoder.reverse,
+                **{
+                    "query": "40.75376406311989, -73.98489005863667",
+                    "exactly_one": True,
+                    "lang": 'ru'
+                }
+            )
+        self.assertEqual(loc.raw['adminName1'], 'Нью-Йорк')
+
+        with self.assertRaises(ValueError):
+            self._make_request(
```

Comment:
Any reason to use the raw `self._make_request` instead of `self.reverse_run`? If no such, please change to `self.reverse_run`.